poly1305/src/fuzz.rs

```diff
@@ -0,0 +1,153 @@
+use super::fuzz_avx2;
```

Comment:
Re: this module, just curious: what fuzzer were you actually using to find these failing cases? Should we add a toplevel `fuzz/` directory?

Reply:
I used [afl.rs](https://github.com/rust-fuzz/afl.rs), which appears to prefer that its fuzzing crate be separate from the main crate (or at least, that's how I interpreted its docs).

Reply:
Do these changes just magically work with afl.rs somehow, or is there an associated config needed to run the fuzzer?

Reply:
[The tutorial](https://rust-fuzz.github.io/book/afl/tutorial.html) covers most of it, but you do need to specify a `main.rs`. I used this:
```
#[macro_use]
extern crate afl;

fn main() {
    fuzz!(|data: &[u8]| {
        // Use first 32 bytes of data as key.
        if data.len() >= 32 {
            poly1305::fuzz_avx2((&data[0..32]).into(), &data[32..]);
        }
    });
}
```
I could inline the length-check and data-splitting into `poly1305::fuzz_avx2` so that the helper here is the most obvious possible thing?

Reply:
Could you throw that code into something like `fuzz/main.rs`?